osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -809,6 +823,125 @@ public override bool OnBackButton()
             return false;
         }
 
+        public void ChangeSpeed(double delta)
+        {
+            ModNightcore modNc = (ModNightcore)((MultiMod)game.AvailableMods.Value[ModType.DifficultyIncrease].First(mod => mod is MultiMod multiMod && multiMod.Mods.Count(modType => modType is ModNightcore) > 0)).Mods.First(mod => mod is ModNightcore);
```

Comment:
This method needs a rewrite.

Reply:
Could you tell me what needs to change for the rewrite so i don't do the same mistake again? (more readable, more compact, ...)